File: Assets/Plugins/iOS/PlayPhoneMultiNet/unity/MNDirectPopupUnity.m
//
//  MNDirectPopupUnity.m
//  PlayPhone SDK for Unity
//
//  Copyright (c) 2012 PlayPhone Inc. All rights reserved.
//

#import "MNDirectPopup.h"

#import "MNDirectPopupUnity.h"

EXTERN_C
void _MNDirectPopup_Init(int actionsBitMask) {
    MARK;

    [MNDirectPopup init:actionsBitMask];
}

EXTERN_C
bool _MNDirectPopup_IsActive() {
    MARK;

    return [MNDirectPopup isActive];
}

EXTERN_C
void _MNDirectPopup_SetActive(bool activeFlag) {
    MARK;

    [MNDirectPopup setActive:activeFlag];
}

File: Assets/Plugins/Android/src/com/playphone/multinet/unity/MNDirectPopupUnity.java
package com.playphone.multinet.unity;

import com.playphone.multinet.*;
import com.unity3d.player.UnityPlayer;

public class MNDirectPopupUnity {
    public static void init(final int actionsBitMask) {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirectPopup.init(actionsBitMask);
            }
        });
    }

    public static boolean isActive() {
        MNUnity.MARK();

        return MNDirectPopup.isActive();
    }

    public static void setActive(final boolean activeFlag) {
        MNUnity.MARK();

        UnityPlayer.currentActivity.runOnUiThread(new Runnable()
        {
            public void run()
            {
                MNDirectPopup.setActive(activeFlag);
            }
        });
    }

}

